src/Orc.Controls/Orc.Controls.Shared/Controls/DateTimePicker/Views/DateTimePickerControl.xaml.cs

```diff
@@ -245,6 +294,209 @@ public override void OnApplyTemplate()
             base.OnApplyTemplate();
 
             AccentColorBrush = TryFindResource("AccentColorBrush") as SolidColorBrush;
+
+            EnableOrDisableHourConverterDependingOnFormat();
+        }
+
+        private void OnFormatChanged()
+        {
+            ApplyFormat();
+        }
+
+        private void ApplyFormat()
```

Comment:
Well.... this works with UI elements. I need to think if we can split this logic to smaller parts.